src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentUtilities.java

```diff
@@ -156,32 +156,35 @@ public static List<ExperimentObservation> importRowsToExperimentObservations(Lis
     }
 
     /**
-     * This method generates a unique key for an observation unit based on the environment and experimental unit ID.
+     * This method generates a unique key for an observation unit based on the environment and experimental unit ID and germplasm GID.
      *
-     * @param importRow the ExperimentObservation object containing the environment and experimental unit ID
+     * @param importRow the ExperimentObservation object containing the environment and experimental unit ID and germplasm GID
      * @return a String representing the unique key for the observation unit
      */
     public static String createObservationUnitKey(ExperimentObservation importRow) {
-        // Extract the environment and experimental unit ID from the ExperimentObservation object
+        // Extract the environment and experimental unit ID and germplasm GID from the ExperimentObservation object
         // and pass them to the createObservationUnitKey method
-        return createObservationUnitKey(importRow.getEnv(), importRow.getExpUnitId());
+        return createObservationUnitKey(importRow.getEnv(), importRow.getExpUnitId(), importRow.getGid());
```

Comment:
Wouldn't env, exp unit id, and gid all be the same for the sub entity case? If there were 5 repeats, ex:
env, exp unit id, gid
CA, 1, 1
CA, 1, 1
CA, 1, 1
CA, 1, 1
CA, 1, 1

We will have sub unit id available once it's added to the download file that would uniquely identify each row.